src/backend/modules/url/url.service.js

```diff
@@ -18,65 +18,62 @@ class UrlServiceImp {
 
     async createOne({ url, slug }, userDetail, urlRecaptcha) {
         let newSlug;
-        await axios({
+        const { data } = await axios({
             method: 'post',
             url: urlRecaptcha
-        })
-            .then(response => response.data)
-            .then(async googleResponse => {
-                if (googleResponse.success === true) {
-                    const keywords = parseUrl(url);
-
-                    if (slug) {
-                        if (this.isInvalidSlug(slug)) {
-                            throw new DuplicateException('Invalid slug');
-                        }
-
-                        const isSlugExisted = await this.repository.findBySlug(slug);
-
-                        if (isSlugExisted) {
-                            throw new DuplicateException(`Slug (${slug}) is already existed`);
-                        }
-
-                        keywords.push(slug);
-                        const newUrl = new Url();
-                        newUrl.slug = slug;
-                        newUrl.url = url;
-                        newUrl.keywords = keywords.map(keyword => keyword.toLowerCase());
-                        newUrl.isCustom = true;
-                        newUrl.userId = userDetail?.id;
-
-                        await this.repository.createOne(newUrl.toJson());
-                        return slug;
-                    }
-
-                    const foundUrl = await this.repository.findRandom(url);
-                    // return if there is a random slug
-                    if (foundUrl && !userDetail && !foundUrl.userId) {
-                        return foundUrl.slug;
-                    }
-
-                    let idLength = DEFAULT_ID_LENGTH;
-
-                    const isLoop = true;
-                    do {
-                        newSlug = generateId(idLength);
-                        // eslint-disable-next-line no-await-in-loop
-                        const isIdExisted = (await this.repository.findBySlug(newSlug));
-                        if (!isIdExisted) break;
-                        idLength += 1;
-                    } while (isLoop);
-
-                    keywords.push(newSlug);
-
-                    const newUrl = new Url();
-                    newUrl.slug = newSlug;
-                    newUrl.url = url;
-                    newUrl.userId = userDetail?.id;
-                    newUrl.keywords = keywords.map(keyword => keyword.toLowerCase());
-                    await this.repository.createOne(newUrl.toJson());
-                } else throw new NotFoundException('Captcha is invalid');
-            });
+        });
+        if (data.success === true) {
```

Comment:
if(!data.success) throw ......
const keywords = ........

This looks much cleaner